EUI 5.8.1 comes with a brand color refresh. Also removes lots of KUI

snide opened the pull request:
## Summary

For team review pings:

- [x] I changed the APM JS variables that mirrored EUI. It'd be nice if you all imported this from the JSON we dist, but it was simple enough.
- [x] Beats team - I changed some hardcoded hex values you had in styled components. Also prefer if you pulled this in from EUI's vars.
- [x] Security - Small template / sass changes to match colors.

### Color refresh

EUI 5.7.0 had a color refresh which switched around our palette to better match branding guidelines. Hex colors are still hard coded in large parts of Kibana so most of the changes not in `kbn/ui-framework` are simple shifts to match that styling.

### Styling of KUI

KUI now imports the EUI global scope. I went through and touched a lot of the global variables in KUI to better match EUI. Outside of the coloring, I did some very minor work around KUI form fields and the verticalRhythm component.

### Removal of many KUI components

KUI remains deprecated since 6.2.0 but still exists. I went ahead and removed any components and styling that were no longer being used in Kibana and then removed several KUI components that were being minimally used by replacing them with EUI styling, using EUI components or selectors instead. This resulted in KUI being a little under half its original size and is the reason this PR has such a large negative number.

### Global nav cleanup

With the color change the global nav icons in the K6 nav was really off. I went ahead reworked them to use EUI icons and cleanup them up stylistically. The static images are still available if no EUI icon is used so this is backwards compatible and should be fine for custom plugins outside of core Kibana.

### Tests

The color changes required me to rebuild the baselines for the TSVB screenshot. We have a lot of hard coded hex values that in turn are converted back and forth between rgb and hex. In one case I chose not to not change an advanced setting default to match our new colors because the tests went a little nuts. It's the green color and is mostly close to what we're using in both palettes. I'd like to do a later sweep there and make these tests less reliant on copy / paste codes, but given FF timing, I'm just avoiding that value for now.

### Updates EUI to 5.8.1 from 5.6.2

## [`5.8.1`](https://github.com/elastic/eui/tree/v5.8.1)

**Note: this release is a backport containing fixes made in `6.0.0`**

**Bug fixes**

- Fixed some EUI services' TS definitions ([#1380](https://github.com/elastic/eui/pull/1380))

## [`5.8.0`](https://github.com/elastic/eui/tree/v5.8.0)

**Note: this release broke some of the exported TypeScript definitions.**

- Reinstate ([#1353](https://github.com/elastic/eui/pull/1353)) `onBlur` action on `EuiComboBox` ([#1364](https://github.com/elastic/eui/pull/1364))
- Convert roughly half of the services to TypeScript ([#1360](https://github.com/elastic/eui/pull/1360))

**Bug fixes**

- Fixed `onCreateOption` callback of `EuiComboBox` so it isn't called when the input is empty ([#1364](https://github.com/elastic/eui/pull/1364))
- Added `anchorClassName` prop to `EuiPopover` ([#1367](https://github.com/elastic/eui/pull/1367))
- Added support for `fullWidth` on `EuiSuperSelect` ([#1367](https://github.com/elastic/eui/pull/1367))
- Applied new scrollbar customization for Firefox ([#1367](https://github.com/elastic/eui/pull/1367))
- Fixed `EuiSuperSelect` from accessing ref when unmounted ([1369](https://github.com/elastic/eui/pull/1369))
- Allow any color value to be passed to `EuiIcon` ([#1370](https://github.com/elastic/eui/pull/1370))

## [`5.7.0`](https://github.com/elastic/eui/tree/v5.7.0)

- Adjust EUI coloring to better match brand guidelines from Creative Services ([#1356](https://github.com/elastic/eui/pull/1356))



### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [x] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)

### For maintainers

- [x] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)
- [x] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

snide committed:
kui eui start

snide committed:
Merge remote-tracking branch 'upstream/master' into kui/vars

snide committed:
eui 5.7.0, along with KUI theming

snide committed:
more cleanup coloring

snide committed:
remove kui bits no longer in use

elasticmachine commented:
Pinging @elastic/kibana-design

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3301/)

snide committed:
deleting more unused kui components

snide committed:
more machete work in kui

snide committed:
more machete work in kui

snide committed:
screen reader fixes

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3313/)

snide commented:
jenkins, test this

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3321/)

cchaos reviewed (commented):
Most of it looks good, some better than others, but these are just a few that I find needs a quick fix:

1. Sidebar

<img width="39" alt="screen shot 2018-12-12 at 14 40 47 pm" src="https://user-images.githubusercontent.com/549577/49895397-50085f00-fe1e-11e8-957b-66305f9720e0.png">

2. Visualize top bar

<img src="https://d.pr/free/i/QxyH1W+" />

3. For some reason the disabled buttons in Kibana look different from EUI

<img width="630" alt="screen shot 2018-12-12 at 15 13 09 pm" src="https://user-images.githubusercontent.com/549577/49896152-834bed80-fe20-11e8-8380-492b9ed7dda5.png">

4. I found a page where the background colors don't line up

<img width="1248" alt="screen shot 2018-12-12 at 15 15 39 pm" src="https://user-images.githubusercontent.com/549577/49896260-d887ff00-fe20-11e8-9f06-ec37a32e7f16.png">

snide committed:
eui icons for kibana left nav

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3538/)

snide committed:
some search and replace hex coloring, adjust apm to match EUI

snide committed:
apm snapshot updates

snide committed:
minor cleanup

snide committed:
Merge remote-tracking branch 'upstream/master' into kui/vars

snide commented on packages/kbn-ui-framework/src/global_styling/variables/_colors.scss:
```diff
@@ -1,52 +1,52 @@
+$kuiLinkColor: $euiColorPrimary;
 $kuiLinkColor-isHover: darken($kuiLinkColor, 10%);
 $kuiInputTextColor: $kuiTextColor;
 $kuiInputBackgroundColor: $kuiColorWhite;
```
KUI has it's own variables for dark mode :/

snide replied:
That said, I need to check more into the dark theme coloring. I see some breaks this introduced.

snide commented on packages/kbn-ui-framework/src/components/vertical_rhythm/_index.scss:
```diff
@@ -1,6 +1,6 @@
-$verticalRhythmSmallSpacing: 5px;
-$verticalRhythmLargeSpacing: 20px;
-$verticalRhythmXLargeSpacing: 40px;
+$verticalRhythmSpacing: $euiSize;
```
Only thing that really scares me is who knows what absolute pixel heights were constructed assuming so sort of constant from these.

I'm not worried about the visual difference though. I based it the off intended usage I saw and mapped against our equivalent. These were used similar to our spacer components and not for pixel precision.

KUI's base was 10 and that one was the most common vertical spacing selector in Kibana. The equiv on our side is EuiSpacer size="m". It's most often used to provide distance between larger blocks and panels. Here's a good example where at least spacing in some mixed views is getting at least some semblance of consistency.

![image](https://user-images.githubusercontent.com/324519/49921218-ba8ebe80-fe61-11e8-89d0-36a13c008ead.png)

`.kuiVerticalRhythmSmall` is mostly used for separation of form elements and checkboxes from what I could see. In all the cases I checked the spacing made sense and was closer to our usage in EUI. Below are some screens showing common usage of the smaller sizes.

![image](https://user-images.githubusercontent.com/324519/49920860-92eb2680-fe60-11e8-85d0-4c2e9dd0d41e.png)

![image](https://user-images.githubusercontent.com/324519/49920864-97afda80-fe60-11e8-81cb-7139e002ccd3.png)

This is all pretty quick damage. Spacing is one of the quick things I think we can change in KUI to make things _feel_ a little closer. For a half minute I thought... hey, why don't I just extend EUI forms into KUI ones, but I backed off because I kind of want to be able to look at the page and still know what needs to be removed. Ultimately, I think we're in the single digit months of how long KUI will actually be around.

This stuff ultimately I think is a band aid.

cchaos replied:
Got it! It's one of those situations where I don't know if it's intentional or not. Looks like it was thoroughly thought through. 🙇‍♀️

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3543/)

snide committed:
Merge remote-tracking branch 'upstream/master' into kui/vars

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3674/)

snide committed:
Merge remote-tracking branch 'upstream/master' into kui/vars

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3717/)

snide committed:
jest

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3732/)

snide committed:
import service directly for now

chrisronline reviewed (approved):
LGTM!

snide committed:
5.8.1 and nav fix

snide committed:
merge master

snide committed:
fix nav menu, get collapse button back in kui

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4255/)

snide committed:
update baseline area chart tests for color

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4264/)

snide committed:
think i found it

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4272/)

snide committed:
update baseline now that color reverted

snide commented:
![image](https://user-images.githubusercontent.com/324519/50132691-35c6ea80-023d-11e9-9ad5-9b5462551c42.png)


Ugh. I updated the baseline, but then changed the color back. I've now updated the baseline again now that it works. I think this should be the last one. Got the rest of the tests to pass now.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4278/)

snide commented:
@cchaos This now passes and is merged up. If you can during your morning, feel free to do a more thorough check. You're welcome to commit directly or just give a PR if you notice anything sizable.

cchaos commented:
I think we should be good to go. I thought about doing some more tweaks in the theming LESS styles, but I've already completely gotten rid of those in my next SASS conversion PR. So we can just wait for those final color conversions in that PR. But I'd really like to get this one in first.

snide committed:
merge master, update snaps

snide commented:
Merged back up to master. Once tests pass I'll be merging this one.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4404/)

snide committed:
snapshots

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4422/)
